scripts/fetch_dependencies.py

```diff
@@ -91,13 +91,18 @@ def UpdateGitHubRepo(repoRootUrl, location, commit):
 
     reqdCommit = commit
 
-    print("\nChecking out commit: %s for %s\n"%(reqdCommit, targetPath))
-
     if os.path.isdir(targetPath):
         # Directory exists - get latest from git using pull.
-        print("Directory " + targetPath + " exists. \n\tUsing 'git pull' to get latest")
-        sys.stdout.flush()
         try:
+            if reqdCommit is not None:
+                currentCommit = subprocess.check_output(["git", "-C", targetPath, "rev-parse", "HEAD"], shell=SHELLARG).decode().strip()
+                if currentCommit == reqdCommit:
+                    print("Directory " + targetPath + " exists and is at expected commit. Nothing to do.")
+                    sys.stdout.flush()
+                    return
+            print("Directory " + targetPath + " exists but is not at the required commit. \n\tUsing 'git pull' to get latest")
+            sys.stdout.flush()
+            subprocess.check_call(["git", "-C", targetPath, "checkout", "master"], shell=SHELLARG)
             subprocess.check_call(["git", "-C", targetPath, "pull", "origin"], shell=SHELLARG)
```

Comment:
I think we can avoid this checkout + pull + checkout combination by replacing the checkout + pull with a 'git -C <targetPath> fetch --tags -f origin'. It seems to work for me and is definitely faster than temporarily swapping to master. If you don't foresee any issues with this alternative, can you try it on your side?

Reply:
Nice. That is indeed a better solution. I've updated the pull-request. Your approach is working for me.